youtube_dl/downloader/external.py

```diff
@@ -200,6 +200,69 @@ def _make_cmd(self, tmpfilename, info_dict):
         return cmd
 
 
+class Aria2pFD(ExternalFD):
+    ''' Aria2pFD class
+    This class support to use aria2p as downloader.
+    (Aria2p, a command-line tool and Python library to interact with an aria2c daemon process
+    through JSON-RPC.)
+    It can help you to get download progress more easily.
+    To use aria2p as downloader, you need to install aria2c and aria2p, aria2p can download with pip.
+    Then run aria2c in the background and enable with the --enable-rpc option.
+    '''
+    try:
+        import aria2p
+        __avail = True
+    except ImportError:
+        __avail = False
+
+    @classmethod
+    def available(cls):
+        return cls.__avail
+
+    def _call_downloader(self, tmpfilename, info_dict):
+        import aria2p
+        aria2 = aria2p.API(
```

Comment:
The `aria2p` module namespace was already imported:
```suggestion
        aria2 = self.aria2p.API(
```